src/inference/tests/unit/cpu_reserve_test.cpp

```diff
@@ -0,0 +1,812 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gtest/gtest.h>
+
+#include <common_test_utils/test_common.hpp>
+
+#include "ie_system_conf.h"
+#include "os/cpu_map_info.hpp"
+
+using namespace testing;
+using namespace ov;
+
+namespace {
+
+#if defined(__linux__) || defined(_WIN32)
+
+struct LinuxCpuReserveTestCase {
+    int _processors;
+    int _sockets;
+    std::vector<std::vector<int>> _proc_type_table;
+    std::vector<std::vector<int>> _cpu_mapping_table;
+    std::vector<std::vector<int>> _streams_info_table;
+    std::vector<std::vector<int>> _stream_processors;
+    int _cpu_status;
+};
+
+class LinuxCpuReserveTests : public CommonTestUtils::TestsCommon,
+                             public testing::WithParamInterface<std::tuple<LinuxCpuReserveTestCase>> {
+public:
+    void SetUp() override {
+        const auto& test_data = std::get<0>(GetParam());
+
+        std::vector<std::vector<int>> test_processors;
+        std::vector<int> test_numa_node_ids;
+
+        ov::reserve_cpu_by_streams_info(test_data._streams_info_table,
+                                        test_data._cpu_mapping_table,
+                                        test_data._proc_type_table,
+                                        test_data._sockets,
+                                        test_processors,
+                                        test_data._cpu_status);
+
+        ASSERT_EQ(test_data._stream_processors, test_processors);
+    }
+};
+
+LinuxCpuReserveTestCase _2sockets_72cores_hyper_36streams = {
+    72,  // param[in]: the number of processors
```

Comment:
as we agreed before it is better to state "logical processors"

Reply:
Ok, done.